src/metrics/metrics.jl

```diff
@@ -279,44 +158,39 @@ end
 #     return rs.outcomes[:relative_loc_taxa_cover]
 # end
 
-relative_loc_taxa_cover = Metric(_relative_loc_taxa_cover, (:timesteps, :taxa, :location, :scenarios))
+relative_loc_taxa_cover = Metric(_relative_loc_taxa_cover, (:timesteps, :taxa, :sites, :scenarios))
 
+"""
+    relative_juveniles(X::AbstractArray{T}, coral_spec::DataFrame)::AbstractArray{T} where {T<:Real}
+    relative_juveniles(rs::ResultSet)::AbstractArray
 
+Juvenile coral cover relative to total site area.
+"""
 function _relative_juveniles(X::AbstractArray{T}, coral_spec::DataFrame)::AbstractArray{T} where {T<:Real}
     # Cover of juvenile corals (< 5cm diameter)
-    juv_groups::AbstractArray{<:Real} = X[species=coral_spec.class_id .== 1] .+ X[species=coral_spec.class_id .== 2]
+    juv_groups = X[species=(coral_spec.class_id .== 1)] .+ X[species=(coral_spec.class_id .== 2)]
```

Comment:
If `X` is a YAXArray then it is possible that indexing like this can error if the values `coral_spec.class_id .== 1` are all false, you would receive a very uninformative error message `ArgumentError: Chunk sizes must be strictly positive`.

My solution was to cast to a matrix/vector however this may be undersirable

Reply:
Two things:

1. This should never be an "all false" vector in this context because we are selecting one of the size classes (if I'm not mistaken).
2. In any case, what do you think of the following? I'm not sure we should use this HERE, because of 1., but in other places where an all false vector might happen this won't raise an error. 

```
species_idx::Vector{Int64} = collect(X.species)
juv_groups = X[species=At(species_idx[coral_spec.class_id .== 1])] .+ X[species=At(species_idx[coral_spec.class_id .== 2])]
```

Reply:
My attempts at using the suggestion all resulted in an error. At this time the only error free solution I can think of is
```julia
bitVec = coral_spec.class_id .== 1
if sum(bitVec) == 0
    juv_groups = X[species=At([])] # results in 0 dimensional array
else 
    # normal behaviour
end
```
However, this doesn't apply to the above code as I assume both terms on either side of the addition would be of equal shape.

The reason I was casting to Matrix/Vector is because they support indexing with a bitVector of all 0's. Either way the occurences I found of it handle the zero case immediately after.

Reply:
Hum, I don't understand why your attempts to use my suggestions resulted in error (what was the error?), I tested here and it worked. In any case, do you think we can leave this this way, since there shouldn't be an all false vector? If, by some mistake, there is an all false vector here, I think it **should** raise an error. Let me know what you think.

Reply:
I am happy for this to go through then. 

The error I received was the same as usual `ArgumentError: Chunk sizes must be strictly positive`. Reproduced with

```julia 
nmy = YAXArray((Dim{:dim1}(1:20), ), rand(20))
idx = collect(nmy.dim1)
nmy[dim1=At(idx[rand(20) .< 0])] # guarantee bit vector is all zeros
```

Reply:
Hum, I see what the problem is. I tested `nmy[dim1=At([])]` and that works (for some reason). But `nmy[dim1=At(Int64[])]` doesn't.